TUTORIAL.md

```diff
@@ -0,0 +1,92 @@
+# Guide to Solving and Reviewing Tweet Shortener
+
+Before we start coding our methods, we need to create a hash where our methods can look up which word needs to be substituted.
+
+```ruby
+def dictionary
+  dictionary = {
+    "too" => "2",
+    "to" => "2",
+    "two" =>"2",
+    "four" => "4",
+    "for" => "4",
+    "be" => "b",
+    "you" => "u",
+    "at" => "@",
+    "and" => "&"
+  }
+end
+```
+
+### `word_substituter`
+
+Let us start with defining our `word_substituter` method. This method will also take in an argument `tweet`.
+
+```ruby
+def word_substituter(tweet)
+end
+```
+After putting a `binding.pry` in our method, we know that the passed in `tweet` is a sting.
+
+We need to split the string to get every word individually, then we need to iterate over every word in our string to compare these words to our `dictionary` hash. If the downcased word is a key in our hash, then we need to replace that word with it substitute, else we need to return that word. Also we need to `join` every word back to a string. 
```

Comment:
Start a bit higher level. Why are you spliting the words individually?

Your higher level goal here is we need to find any words that we know we can shorten. Problem is we are given a string. If I was doing this as a human, I would just go word by word looking for the words I need to replace. How can I do that, If I google "iterate over words string ruby" and I notice the `#split` method that converts a string into an array!

So we are going to convert our string into an array, then use some stuff we already know to iterate over a standard array. While in there I need to check each word, and then replace. We downcase so that case doesn't matter. After all this iteration I'm given an array of words, but our method is supposed to return a string. So after googling "convert array to string ruby" I find the `#join` method.